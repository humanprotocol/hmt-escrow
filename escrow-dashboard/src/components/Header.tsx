import * as React from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Link from '@mui/material/Link';
import Typography from '@mui/material/Typography';
import Toolbar from '@mui/material/Toolbar';

import humanLogo from 'src/assets/logo.svg';
import TokenPriceIndicator from './TokenPriceIndicator';

const Header: React.FC = (): React.ReactElement => (
  <Box sx={{ flexGrow: 1 }}>
    <AppBar position="fixed" sx={{ background: '#fff', boxShadow: 'none' }}>
      <Toolbar disableGutters>
        <Box
          sx={{
            display: 'flex',
            flexDirection: 'row',
            justifyContent: 'space-between',
            alignItems: 'center',
            width: '100%',
            padding: {
              xs: '30px 8px 26px',
              sm: '30px 28px 26px',
              md: '30px 52px 26px',
            },
          }}
        >
          <Box display="flex" alignItems="center">
            <img src={humanLogo} alt="logo" />
            <Typography variant="h6" color="primary" ml="10px">
              Scan
            </Typography>
          </Box>
          <Box display="flex" alignItems="center" ml="auto">
            <TokenPriceIndicator />
            <Link
              href="https://humanprotocol.org"
              target="_blank"
              sx={{
                textDecoration: 'none',
                ml: '28px',
                display: { xs: 'none', sm: 'block' },
              }}
            >
              <Typography variant="body2" fontWeight={600}>
                HUMAN Website
              </Typography>
            </Link>
          </Box>
        </Box>
      </Toolbar>
    </AppBar>
  </Box>
);

export default Header;
