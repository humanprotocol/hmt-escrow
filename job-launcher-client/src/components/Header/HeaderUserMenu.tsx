import * as React from 'react';
import { AppBar, Box, IconButton, Toolbar } from '@mui/material';
import { styled } from '@mui/material/styles';
import { ConnectButton } from '@rainbow-me/rainbowkit';

// icons
import { RiMenu3Line } from 'react-icons/ri';
// import { BiSearch } from 'react-icons/bi';
import { useSelector } from 'react-redux';
import AccountMenu from './AccountMenu';
import { makeIsTokenAuthenticated } from '../../services/redux/selectors/auth';
import { drawerWidth } from '../Layouts/DashboardLayout';

const AppBarStyle = styled(AppBar)(({ theme }: any) => ({
  boxShadow: 'none',
  backdropFilter: 'blur(6px)',
  backgroundColor: 'rgba(255, 255, 255, 0.72)',
  color: '#333333',
  [theme.breakpoints.up('sm')]: {
    width: `calc(100% - ${drawerWidth}px)`,
    flexShrink: 0,
  },
}));

const ToolbarStyle = styled(Toolbar)(() => ({
  display: 'flex',
  justifyContent: 'space-between',
  alignContent: 'flex-start',
  alignItems: 'center',
}));

const ContainerStyle = styled(Box)(({ theme }: any) => ({
  display: 'grid',
  gap: theme.spacing(0.5),
  gridAutoFlow: 'column',
  alignItems: 'center',
}));

const ToggleButtonStyle = styled(IconButton)(({ theme }: any) => ({
  [theme.breakpoints.up('sm')]: {
    display: 'none',
  },
}));

const HeaderUserMenu = ({ onClick }: any) => {
  const isAuth = useSelector(makeIsTokenAuthenticated);

  return (
    <AppBarStyle position="fixed">
      <ToolbarStyle>
        <ContainerStyle>
          <ToggleButtonStyle
            color="inherit"
            aria-label="open drawer"
            edge="start"
            onClick={onClick}
          >
            <RiMenu3Line />
          </ToggleButtonStyle>

          {/* <IconButton aria-label="search">
            <BiSearch fontSize="small" />
          </IconButton> */}
        </ContainerStyle>
        {isAuth && (
          <ContainerStyle>
            <ConnectButton />
            <AccountMenu />
          </ContainerStyle>
        )}
      </ToolbarStyle>
    </AppBarStyle>
  );
};

export default HeaderUserMenu;
