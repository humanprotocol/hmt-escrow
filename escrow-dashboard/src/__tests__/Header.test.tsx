import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import Header from 'src/components/Header';

describe('when rendered Header component', () => {
  beforeAll(async () => {
    global.fetch = jest.fn().mockImplementationOnce(() =>
      Promise.resolve({
        status: 200,
        json: () =>
          Promise.resolve({
            market_data: {
              circulating_supply: 0,
              total_supply: 0,
              current_price: { usd: 0 },
              price_change_percentage_24h: 0,
            },
          }),
      })
    );
  });

  it('should render `text` prop', async () => {
    await act(async () => {
      render(<Header />);
    });
    expect(screen.getByText(/HUMAN Website/)).toBeInTheDocument();
  });
});

it('Header component renders correctly, corresponds to the snapshot', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
