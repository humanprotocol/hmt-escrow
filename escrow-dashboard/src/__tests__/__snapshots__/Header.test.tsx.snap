// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly, corresponds to the snapshot 1`] = `
<div
  className="MuiBox-root css-i9gxme"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-164jy24-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
    >
      <div
        className="MuiBox-root css-1qtwkyj"
      >
        <div
          className="MuiBox-root css-70qvj9"
        >
          <img
            alt="logo"
            src="test-file-stub"
          />
          <h6
            className="MuiTypography-root MuiTypography-h6 css-i6uuzy-MuiTypography-root"
          >
            Scan
          </h6>
        </div>
        <div
          className="MuiBox-root css-84nudp"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-q1xa4c-MuiTypography-root-MuiLink-root"
            href="https://humanprotocol.org"
            onBlur={[Function]}
            onFocus={[Function]}
            target="_blank"
          >
            <p
              className="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            >
              HUMAN Website
            </p>
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
